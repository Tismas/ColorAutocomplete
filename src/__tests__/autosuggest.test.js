import React from 'react';
import Autosuggest from '../components/Autosuggest';
import renderer from 'react-test-renderer';

test('With empty input', () => {
  const component = renderer.create(
    <Autosuggest 
        value=""
        suggestions={[
            {name: "Adam"}, 
            {name: "Sija"}, 
            {name: "Paweł"}
        ]}

        onChange={()=>{}}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('With not empty input and possible suggestions', () => {
  const component = renderer.create(
    <Autosuggest 
        value="Ad"
        suggestions={[
            {name: "Adam"}, 
            {name: "Sija"}, 
            {name: "Paweł"}
        ]}

        onChange={()=>{}}
    />,
  );
  let tree = component.toJSON();
  tree.children[0].props.onChange({target:{value:'Ad'}});
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('With not empty input and possible suggestions', () => {
  const component = renderer.create(
    <Autosuggest 
        value="ij"
        suggestions={[
            {name: "Adam"}, 
            {name: "Sija"}, 
            {name: "Paweł"}
        ]}

        onChange={()=>{}}
    />,
  );
  let tree = component.toJSON();
  tree.children[0].props.onChange({target:{value:'ij'}});
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('With not empty input and no possible suggestions', () => {
  const component = renderer.create(
    <Autosuggest 
        value="Szy"
        suggestions={[
            {name: "Adam"}, 
            {name: "Sija"}, 
            {name: "Paweł"}
        ]}

        onChange={()=>{}}
    />,
  );
  let tree = component.toJSON();
  tree.children[0].props.onChange({target:{value:'Szy'}});
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});