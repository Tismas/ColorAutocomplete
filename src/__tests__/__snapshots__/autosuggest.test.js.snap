// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With empty input 1`] = `
<div
  className="autosuggest-wrapper"
>
  <input
    className="autosuggest-input "
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`With not empty input and no possible suggestions 1`] = `
<div
  className="autosuggest-wrapper"
>
  <input
    className="autosuggest-input "
    onChange={[Function]}
    value="Szy"
  />
</div>
`;

exports[`With not empty input and possible suggestions 1`] = `
<div
  className="autosuggest-wrapper"
>
  <input
    className="autosuggest-input no-bottom-border-radius"
    onChange={[Function]}
    value="Ad"
  />
  <div
    className="autosuggest-suggestions-container"
  >
    <div
      className="autosuggest-suggestion"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <span>
        
      </span>
      <b>
        Ad
      </b>
      <span>
        am
      </span>
    </div>
  </div>
</div>
`;

exports[`With not empty input and possible suggestions 2`] = `
<div
  className="autosuggest-wrapper"
>
  <input
    className="autosuggest-input no-bottom-border-radius"
    onChange={[Function]}
    value="ij"
  />
  <div
    className="autosuggest-suggestions-container"
  >
    <div
      className="autosuggest-suggestion"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <span>
        S
      </span>
      <b>
        ij
      </b>
      <span>
        a
      </span>
    </div>
  </div>
</div>
`;
